referenceframe/frame_system.go

```diff
@@ -712,3 +712,18 @@ func poseFromPositions(frame Frame, positions map[string][]Input) (spatial.Pose,
 	}
 	return frame.Transform(inputs)
 }
+
+// AddFrameBetween takes an existing framesystem which has general form:
+// originalFS := world - f1 - f2 - ...
+// and places the middle frame between f1 and world resulting in
+// newFS := world - middle - f1 - f2 ...
+func AddFrameBetween(middle Frame, originalFS FrameSystem) (FrameSystem, error) {
```

Comment:
This seems wrong.

This will offset an *entire* framesystem. "AddFrameBetween" makes me think that it should be able to be inserted essentially as a new parent of some arbitrary frame, affecting only that frame and its children. This would be more useful functionality.

Reply:
Will do now and push up changes shortly.

Reply:
fixing this now.

Reply:
After some thought, I think I will just remove this function. It would only really be used in one place, within `createBaseMoveRequest` of `move_request.go`.

Reply:
For posterity, here is what I believe the function should look like per Peter's comment.
```
// ToDo: add comment
func AddFrameBetween(insert, firstChild Frame, originalFS FrameSystem) (FrameSystem, error) {
	newFS := NewEmptyFrameSystem("augmented-FS")
	if err := newFS.AddFrame(insert, newFS.World()); err != nil {
		return nil, err
	}
	fsSubset, err := originalFS.FrameSystemSubset(firstChild)
	if err != nil {
		return nil, err
	}
	if err := newFS.MergeFrameSystem(fsSubset, insert); err != nil {
		return nil, err
	}
	return newFS, nil
}
```